src/nanoemoji/nanoemoji.py

```diff
@@ -190,6 +195,77 @@ def _not_impl(*_):
     raise NotImplementedError("%s not implemented" % FLAGS.color_format)
 
 
+def _draw(source: SVGPath, dest: Glyph, svg_units_to_font_units: Affine2D):
+    pen = TransformPen(dest.getPen(), svg_units_to_font_units)
+    skia_path(source.as_cmd_seq()).draw(pen)
+
+
+def _next_name(ufo: ufoLib2.Font, name_fn) -> str:
+    i = 0
+    while name_fn(i) in ufo:
+        i += 1
+    return name_fn(i)
+
+
+def _glyph(color_glyph: ColorGlyph, painted_layer: PaintedLayer) -> Glyph:
+    ufo = color_glyph.ufo
+
+    glyph = ufo.newGlyph(_next_name(ufo, lambda i: f"{color_glyph.glyph_name}.{i}"))
+    glyph.width = ufo.info.unitsPerEm
+
+    svg_units_to_font_units = color_glyph.transform_for_font_space()
+
+    if painted_layer.reuses:
+        # Shape repeats, form a composite
+        composite_base = ufo.newGlyph(
+            _next_name(ufo, lambda i: f"{glyph.name}.component.{i}")
+        )
+
+        _draw(painted_layer.path, composite_base, svg_units_to_font_units)
+
+        glyph.components.append(
+            Component(baseGlyph=composite_base.name, transformation=Affine2D.identity())
+        )
+
+        for transform in painted_layer.reuses:
+            # We already redrew the component into font space, don't redo it
+            # scale x/y translation and flip y movement to match font space
+            transform = transform._replace(
```

Comment:
i'm thinking if instead of directly manipulating the inner matrix values with a namedtuple `_replace`, we could use matrix products here?
The operation you're doing here is equivalent to: 

```
transform = Affine2D.product(svg_units_to_font_units.inverse(), Affine2D.product(transform, svg_units_to_font_units))
```

That is, first right-multiply `svg_units_to_font_units` by the reuse `transform` (which refers to SVG space) to obtain a map from untransformed SVG units to final font units; then since we have already applied `svg_units_to_font_units` when we drew the base glyph and don't want to apply that twice, we right-multiply the result by the `inverse()` of the `svg_units_to_font_units` and obtain our final `transform` that now only has the transformed translation vector from base to component in font units.

If that confuses/complicates things further, feel free to ignore.

Reply:
I confess to finding that considerably less clear. `transform.scale *= svg2font.scale` would be nice but I don't know how to do that in python.